### ISSUE TITLE
Let Porymap read layout size from files instead having hardcoded values

### ISSUE THREAD

**Issue by Jaizu** [huderlem/porymap#397]
Atm Porymap will only support the default vanilla dimensions.
It should read the constants by default, and, if those are not found because the project isn't updated, read the hardcoded ones.
```
#define MAP_WIDTH 28
#define MAP_HEIGHT 18
#define MAPCURSOR_X_MIN 1
#define MAPCURSOR_Y_MIN 
#define MAPCURSOR_X_MAX (MAPCURSOR_X_MIN + MAP_WIDTH - 1)
#define MAPCURSOR_Y_MAX (MAPCURSOR_Y_MIN + MAP_HEIGHT - 1)
```

vs

```
    const int padLeft   = 1;
    const int padRight  = 3;
    const int padTop    = 2;
    const int padBottom = 3;
```

**Comment by GriffinRichards** [comment]
> Atm Porymap will only support the default vanilla dimensions.
This is only true for the margin sizes. Users can change the region map dimensions by modifying the `region_map_dimensions` field in `porymap.cfg` (which isn't to say that Porymap shouldn't read the constants instead).

**Comment by Jaizu** [comment]
Welp I think porymap shouldn't have hardcoded values for those. Do you want to suggest a better title for this issue? I don't mind changing it to make it more clear.

**Comment by GriffinRichards** [comment]
I agree, I just wanted to make clear to anyone reading this that you can currently change the size of the region map in Porymap.

"Read region map size from project"?

**Comment by garakmon** [comment]
resolved as of d14bcdbdc19d425ac17fd3487c18e0bd2fd5aeff
